### PR TITLE
/insights

### LINKED ISSUE #6507: [CAL-803] /insights
figma: https://www.figma.com/file/ZJNAFHCAq6rnuGRgUFxjzQ/Cal.com-Teams?node-id=514%3A80787&t=h4kM3EyGKfIPY9o8-4

CleanShot 2023-01-16 at 14 51 05@2x

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-803](https://linear.app/calcom/issue/CAL-803/analytics)</sub>

### PR DESCRIPTION
**Pull request by PeerRich**
- [x] test https://www.tremor.so/
- [x] add upgrade screen
- [x] componentise 
- [x] add data @alannnc @sean-brydon 

CleanShot 2023-01-16 at 16 40 07@2x

Update:
image

### CODE REVIEW

File: apps/web/pages/_app.tsx [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import "@tremor/react/dist/esm/tremor.css";
```
Comment (PeerRich) [inline apps/web/pages/_app.tsx:1]: I wonder if we want to fork this CSS file and apply our own styling

File: apps/web/pages/insights/index.tsx [hunk 2]
```diff
@@ -0,0 +1,437 @@
+import {
```
Comment (PeerRich) [inline apps/web/pages/insights/index.tsx:1]: i am still testing which is why every component is still in the same file to

File: apps/web/pages/insights/index.tsx [hunk 3]
```diff
@@ -0,0 +1,446 @@
...
+  LineChart,
+  Title,
+  Table,
+  TableRow,
+  TableCell,
+  TableHead,
+  TableHeaderCell,
+  TableBody,
+  MultiSelectBox,
+  MultiSelectBoxItem,
```
Comment (sean-brydon) [inline apps/web/pages/insights/index.tsx:19]: This multiselect here is actually better than our multiselect @emrysal @Jaibles https://www.tremor.so/

Comment (PeerRich) [reply apps/web/pages/insights/index.tsx:19]: agree. we could potentially fork tremor or somehow overwrite their styles to make it look even more than ours

File: apps/web/package.json [hunk 4]
```diff
@@ -60,6 +60,7 @@
     "@stripe/react-stripe-js": "^1.10.0",
     "@stripe/stripe-js": "^1.35.0",
     "@tanstack/react-query": "^4.3.9",
+    "@tremor/react": "^2.0.0",
```
Comment (alannnc) [inline apps/web/package.json:63]: Latest tremor version

File: packages/prisma/package.json [hunk 5]
```diff
@@ -16,7 +16,8 @@
     "generate-schemas": "prisma generate && prisma format",
     "post-install": "yarn generate-schemas",
     "seed-app-store": "ts-node --transpile-only ./seed-app-store.ts",
-    "delete-app": "ts-node --transpile-only ./delete-app.ts"
+    "delete-app": "ts-node --transpile-only ./delete-app.ts",
+    "seed-analytics": "ts-node --transpile-only ./seed-analytics.ts"
```
Comment (alannnc) [inline packages/prisma/package.json:20]: Running this command we can inject a test user admin account with a team and bookings in the past. We can run it multiple times to add more bookings to DB.

**Comment by zomars** [comment]
Reviewing

File: apps/console
Comment (zomars) [inline apps/console:1]: Same with this

**Comment by zomars** [comment]
- Missing code self-review
- Missing proper testing instructions

File: apps/web/pages/insights/AverageEventDurationChart.tsx
Comment (zomars) [inline apps/web/pages/insights/AverageEventDurationChart.tsx:1]: These files should not live here. These are making this a valid route http://localhost:3000/insights/AverageEventDurationChart

when it shouldn't

File: apps/web/pages/insights/components/KPICard.tsx
Comment (zomars) [inline apps/web/pages/insights/components/KPICard.tsx:1]: Components should not live inside pages

**Review by zomars — CHANGES_REQUESTED** [review]
Blocking to make some improvements 🙏🏽

**Comment by zomars** [comment]
I see we use "analytics" and "insights" interchangeably. I propose we stick to "insights" everywhere.

**Comment by PeerRich** [comment]
> I see we use "analytics" and "insights" interchangeably. I propose we stick to "insights" everywhere.

agree! Insights everywhere

**Review by zomars — APPROVED** [review]
- Moved most of the code to `features/insights`
- Renamed all `analytics` to `insights`
- Extracted `InsightsProvider`
- Extracted trpc router
- Type fixes

File: packages/features/insights/context/FiltersProvider.tsx [hunk 6]
```diff
@@ -0,0 +1,47 @@
+import { useState } from "react";
+
+import dayjs from "@calcom/dayjs";
+
+import type { FilterContextType } from "./provider";
+import { FilterProvider } from "./provider";
+
+export function FiltersProvider({ children }: { children: React.ReactNode }) {
+  // TODO: Sync insight filters with URL parameters
```
Comment (zomars) [inline packages/features/insights/context/FiltersProvider.tsx:9]: Non blocking but should be made in a follow up

**Comment by PeerRich** [comment]
oh shit, not sure if this should merge this quickly

**Pull request merged**